### PR TITLE
Improves seq2seq example

### PR DESCRIPTION
**Pull request by andsteing**
This PR does a couple of things (see individual commit messages for details)

1. Removes deprecated @jax.mask transformation.
2. Makes seq2seq use Optax.
3. Adds CLU metric logging.
4. Adds Colab.

Note that some - but not all - of the issues in #571 are resolved with this PR.

Also resolves example for #1053

### CODE REVIEW

**Review by levskaya — APPROVED** [review]
Thank you!!!
